cluster-autoscaler/simulator/clustersnapshot/test_utils.go

```diff
@@ -53,3 +57,48 @@ func InitializeClusterSnapshotOrDie(
 		}
 	}
 }
+
+// CreateTestNodesWithPrefix creates n test Nodes with the given name prefix.
+func CreateTestNodesWithPrefix(prefix string, n int) []*apiv1.Node {
+	nodes := make([]*apiv1.Node, n, n)
+	for i := 0; i < n; i++ {
+		nodes[i] = test.BuildTestNode(fmt.Sprintf("%s-%d", prefix, i), math.MaxInt, math.MaxInt)
+		test.SetNodeReadyState(nodes[i], true, time.Time{})
+	}
+	return nodes
+}
+
+// CreateTestNodes creates n test Nodes.
+func CreateTestNodes(n int) []*apiv1.Node {
+	return CreateTestNodesWithPrefix("n", n)
+}
+
+// CreateTestPodsWithPrefix creates n test Pods with the given name prefix.
+func CreateTestPodsWithPrefix(prefix string, n int) []*apiv1.Pod {
+	pods := make([]*apiv1.Pod, n, n)
+	for i := 0; i < n; i++ {
+		pods[i] = test.BuildTestPod(fmt.Sprintf("%s-%d", prefix, i), 1, 1)
+	}
+	return pods
+}
+
+// CreateTestPods creates n test Pods.
+func CreateTestPods(n int) []*apiv1.Pod {
+	return CreateTestPodsWithPrefix("p", n)
+}
+
+// AssignTestPodsToNodes assigns test pods to test nodes based on their index position.
```

Comment:
I would probably rename this comment to

// AssignTestPodsToNodes distributes test pods evenly across test nodes